src/AstroImages.jl

```diff
@@ -85,9 +85,21 @@ for n in (8, 16, 32, 64)
     end
 end
 
+mutable struct properties
+    rgb_image::AbstractArray
```

Comment:
https://docs.julialang.org/en/v1/manual/performance-tips/index.html#Avoid-containers-with-abstract-type-parameters-1

Reply:
This stores a colorview as a MultiMapped array and you told about AbstractArray on riot, so I wrote this as you mentioned.

Reply:
I didn't tell you to do that, just noted that the type is subtype of AbstractArray. What you do in this case is to put the type as a parameter of the type, and the parameter is a subtype of AbstractArray

Reply:
do you mean something like this?
```julia
mutable struct properties{T<:AbstractArray}
    rgb_image::T
....
....
```

Reply:
Also, could you explain a bit if this creates a dispatch or how it will be different?

Reply:
I linked to the wrong section above: https://docs.julialang.org/en/v1/manual/performance-tips/index.html#Avoid-fields-with-abstract-type-1

Reply:
Seriously, just read the whole page, I think I linked most sections so far